olga/analytics/views.py

```diff
@@ -2,38 +2,111 @@
+stream_handler.setLevel(logging.INFO)
 
 
 @method_decorator(csrf_exempt, name='dispatch')
-class ReceiveData(View):
+class AccessTokenRegistration(View):
     """
-    Receives and processes data from the remote edx-platform.
+    Provide access token registration functionality.
+    """
+
+    @staticmethod
+    def registry_a_token_and_return_it():
+        """
+        Create UUID based access token ans store it in database.
+
+        Return same access token.
+        """
+        access_token = uuid.uuid4().hex
+        EdxInstallation.objects.create(access_token=access_token)
 
-    If the platform has already registered, a normal data exchange happens.
-    Otherwise generates a secret token, (registers)saves it to DB with the edx-platform's incoming URL
-    and sends newly generated token to the edx-platform for further
-    data interchange abilities with the server.
+        return access_token
+
+    def post(self, request):  # pylint: disable=unused-argument
+        """
+        Receive primary edX installation request for getting access for dispatch statistics via token.
+
+        Returns HTTP-response with status 201, that means object (installation token) was successfully created.
+        """
+        access_token = self.registry_a_token_and_return_it()
+
+        logger.info('OLGA acceptor registered edX installation with token %s', access_token)
+        return JsonResponse({'access_token': access_token}, status=httplib.CREATED)
+
+
+@method_decorator(csrf_exempt, name='dispatch')
+class AccessTokenAuthorization(View):
+    """
+    Provide access token authorization functionality.
+    """
+
+    @staticmethod
+    def post(request):
+        """
+        Verify that installation is allowed access to dispatch installation statistics.
+
+        Returns HTTP-response with status 200, that means object (installation) with received token exists.
+        Returns HTTP-response with status 401 and refreshed access token, that means object (installation) with
+        received token does not exist and edX installation need to get new one,
+        """
+        access_token_serializer = AccessTokenForm(request.POST)
+
+        if access_token_serializer.is_valid():
+            access_token = request.POST.get('access_token')
+
+            try:
+                EdxInstallation.objects.get(access_token=access_token)
+
+                logger.info('edX installation with token %s was successfully authorized', access_token)
+                return HttpResponse(status=httplib.OK)
+
+            except EdxInstallation.DoesNotExist:
+                logger.info(
+                    'edX installation has no corresponding data in OLGA acceptor database (no received token).'
+                    'Received token is %s.', access_token
+                )
+
+                access_token = AccessTokenRegistration.registry_a_token_and_return_it()
+
+                logger.info('Refreshed token for edX installation is %s', access_token)
```

Comment:
> This should be part of registry_a_token_and_return_it function

It cannot be part of `registration method`, because flow uses `registration method` in two ways: create token and refresh token. Loggers are different for that.